File: Java/Kth-Largest-Element.java
class Solution {
    //param k : description of k
    //param numbers : array of numbers
    //return: description of return
    public int FindKth(ArrayList<Integer> num, int k, int start, int end) {
        if (end == start) return num.get(start);
        int temp = num.get(start);
        int left = start + 1, right = end;
        while (left <= right) {
            while (left <= right && num.get(left) < temp) left ++;
            while (left <= right && num.get(right) >= temp) right --;
            if (left < right) {
                int cnt = num.get(left);
                num.set(left, num.get(right));
                num.set(right, cnt);
                left ++;
                right --;
            }
        }
        num.set(start, num.get(right));
        num.set(right, temp);
        if (right - start + 1 == k) return num.get(right);
        else if(right - start + 1 < k) return FindKth(num, k - (right - start + 1), right + 1, end);
        else return FindKth(num, k, start, right - 1);
    }
    public int kthLargestElement(int k, ArrayList<Integer> numbers) {
        // write your code here
        return FindKth(numbers, numbers.size() - k + 1, 0, numbers.size() - 1);
    }
};

File: Python/Kth-Largest-Element.py
class Solution:
    # @param k & A a integer and an array
    # @return ans a integer
    def FindKth(self, k, A, start, end):
        if start == end:
            return A[start]
        left = start + 1
        right = end
        while left <= right:
            while left <= right and A[right] >= A[start]:
                right -= 1
            while left <= right and A[left] < A[start]:
                left += 1
            if left < right:
                temp = A[left]
                A[left] = A[right]
                A[right] = temp
        temp = A[start]
        A[start] = A[right]
        A[right] = temp
        if right - start + 1 == k:
            return A[right]
        elif right - start + 1 < k:
            return self.FindKth(k - (right - start + 1), A, right + 1, end)
        else:
            return self.FindKth(k, A, start, right - 1)
    def kthLargestElement(self, k, A):
        return self.FindKth(len(A) - k + 1, A, 0, len(A) - 1)